src/components/views/voip/CallView.tsx

```diff
@@ -6,26 +6,46 @@ SPDX-License-Identifier: AGPL-3.0-only OR GPL-3.0-only OR LicenseRef-Element-Com
 Please see LICENSE files in the repository root for full details.
 */
 
-import React, { type FC, useContext, useEffect, type AriaRole, useCallback } from "react";
+import React, { type FC, useContext, useEffect, type AriaRole, useCallback, useRef } from "react";
 
 import type { Room } from "matrix-js-sdk/src/matrix";
-import { type Call, ConnectionState, ElementCall } from "../../../models/Call";
-import { useCall } from "../../../hooks/useCall";
+import { type Call, CallEvent } from "../../../models/Call";
 import MatrixClientContext from "../../../contexts/MatrixClientContext";
 import AppTile from "../elements/AppTile";
 import { CallStore } from "../../../stores/CallStore";
 import { SdkContextClass } from "../../../contexts/SDKContext";
+import { useTypedEventEmitter } from "../../../hooks/useEventEmitter";
+import { useCall } from "../../../hooks/useCall";
 
 interface JoinCallViewProps {
     room: Room;
     resizing: boolean;
     call: Call;
     skipLobby?: boolean;
     role?: AriaRole;
+    onClose: () => void;
 }
 
-const JoinCallView: FC<JoinCallViewProps> = ({ room, resizing, call, skipLobby, role }) => {
+const JoinCallView: FC<JoinCallViewProps> = ({ room, resizing, call, skipLobby, role, onClose }) => {
     const cli = useContext(MatrixClientContext);
+    const mounted = useRef(true);
+    useEffect(
+        () => () => {
+            mounted.current = false;
+        },
+        [],
+    );
+
+    useTypedEventEmitter(
+        call,
+        CallEvent.Presentation,
+        useCallback(
+            (presented: boolean) => {
+                if (!presented && mounted.current) onClose();
+            },
+            [onClose],
+        ),
+    );
```

Comment:
I think this can also get comment phrasing out the logic and why just this is enough.

Reply:
It got even simpler now ('Close' action → onClose is fired)